astronomer/providers/amazon/aws/operators/redshift_data.py

```diff
@@ -43,6 +43,27 @@ def execute(self, context: Context) -> None:  # type: ignore[override]
         if response.get("status") == "error":
             self.execute_complete(context, event=response)
         context["ti"].xcom_push(key="return_value", value=query_ids)
+
+        still_running = False
+        for qid in query_ids:
+            resp = redshift_data_hook.conn.describe_statement(Id=qid)
+            status = resp["Status"]
+            if status == "FAILED":
+                err_msg = f"Error: {resp['QueryString']} query Failed due to {resp['Error']}"
+                msg = f"context: {context}, error message: {err_msg}"
+                raise AirflowException(msg)
+            elif status == "ABORTED":
+                err_msg = "The query run was stopped by the user."
+                msg = f"context: {context}, error message: {err_msg}"
+                raise AirflowException(msg)
+            elif status in ("SUBMITTED", "PICKED", "STARTED"):
```

Comment:
I am sorry but I'm rethinking this. It is good that we're breaking early when we see query is still running. But I think your early approach is still needed to check the count of Finished queries. So we would need another elif condition to also add the Finished queries to check later that the count of finished queries is equal to count of query_ids. The reason I am saying we need this is because Redshift might introduce a new intermediate state later that we don't include here and it would give us a false positive because we won't check that state and then `still_running` would remain False for such a new Intermediate state.

Reply:
yep, that's indeed something we should consider. Thanks for your suggestion! Just updated it